src/compact_bytestrings.rs

```diff
@@ -556,6 +564,7 @@ impl Index<usize> for CompactBytestrings {
 /// assert_eq!(iter.next(), Some(b"Three".as_slice()));
 /// assert_eq!(iter.next(), None);
 /// ```
+#[must_use = "Iterators are lazy and must be consumed to be useful"]
```

Comment:
Please change it to "iterators are lazy and do nothing unless consumed" to be consistent with std.